### PR TITLE
Escape keywords in imports and canonical class names

### LINKED ISSUE #326: Keywords should be escaped in imports and canonical class names
The following will fail and generate invalid code because of reserved words usage:

Case 1 (single import):
```kotlin
  @Test fun singleImportEscapeKeywords() {
    val source = FileSpec.builder("com.squareup.tacos", "Taco")
            .addType(TypeSpec.classBuilder("Taco")
                    .addProperty("madeFreshDate", ClassName("com.squareup.is.fun.in", "Date"))
                    .build())
            .build()
    assertThat(source.toString()).isEqualTo("""
        |package com.squareup.tacos
        |
        |import com.squareup.`is`.`fun`.`in`.Date
        |
        |class Taco {
        |    val madeFreshDate: Date
        |}
        |""".trimMargin())
  }
```

Case 2 (conflicting imports):
```kotlin
@Test fun conflictingImportsEscapeKeywords() {
    val source = FileSpec.builder("com.squareup.tacos", "Taco")
            .addType(TypeSpec.classBuilder("Taco")
                    .addProperty("madeFreshDate1", ClassName("com.squareup.is.fun.in", "Date"))
                    .addProperty("madeFreshDate2", ClassName("com.squareup.do.val.var", "Date"))
                    .build())
            .build()
    assertThat(source.toString()).isEqualTo("""
        |package com.squareup.tacos
        |
        |import com.squareup.`is`.`fun`.`in`.Date
        |
        |class Taco {
        |    val madeFreshDate1: Date
        |
        |    val madeFreshDate2: com.squareup.`do`.`val`.`var`.Date
        |}
        |""".trimMargin())
  }
```
The issue breaks compilation of our project as we have `is` in the package name.

Please read https://github.com/permissions-dispatcher/PermissionsDispatcher/issues/437 for more details about the specific case.

### PR DESCRIPTION
**Pull request by AlexKorovyansky**
Resolves #326

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/Util.kt [hunk 1]
```diff
@@ -127,6 +127,9 @@ internal fun stringLiteralWithQuotes(value: String): String {
   }
 }
 
+internal fun escapeKeywords(canonicalName: String) =
+        canonicalName.split('.').joinToString(".") { escapeIfKeyword(it) }
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/Util.kt]: We use 2-space indents (4-space for continuation indents), could you please format your code accordingly?

File: src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt [hunk 2]
```diff
@@ -268,6 +268,44 @@ class FileSpecTest {
...
+        |
+        |import com.squareup.`is`.`fun`.`in`.Date
+        |
+        |class Taco {
+        |    val madeFreshDate: Date
+        |}
+        |""".trimMargin())
+  }
+
+  @Test fun doubleImportEscapeKeywords() {
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt]: I don't feel like this test adds anything to the `singleImportEscapeKeywords()`, that one should be enough

**Comment by AlexKorovyansky** [comment]
Okay, will make updates

**Comment by Egorand** [comment]
Thanks! Can you please fix the indentation in tests as well?

**Comment by AlexKorovyansky** [comment]
👍

**Pull request merged**